src/plugins/columnSorting/columnSorting.js

```diff
@@ -225,7 +225,7 @@ class ColumnSorting extends BasePlugin {
 
     if (sortPossible) {
       const translateColumnToPhysical = ({ column: visualColumn, ...restOfProperties }) =>
-        ({ column: this.hot.toPhysicalColumn(visualColumn), ...restOfProperties });
+        ({ column: this.hot.renderedToPhysicalColumn(visualColumn), ...restOfProperties });
```

Comment:
I don't see why this should be translated from `rendered` indexes. `sortConfig` should be consistent with any other public API and use visual indexes. Even if some columns are hidden they can be sorted from API, or sorted prior to hiding them.